jbmc/unit/java_bytecode/java_bytecode_convert_method/convert_method.cpp

```diff
@@ -518,3 +546,54 @@ SCENARIO(
     }
   }
 }
+
+SCENARIO(
+  "java convert method variable",
+  "[core][java_bytecode][java_bytecode_convert_method][variable]")
+{
+  symbol_tablet symbol_table;
+  java_string_library_preprocesst string_preprocess;
+  const class_hierarchyt class_hierarchy;
+  java_bytecode_convert_methodt converter{symbol_table,
+                                          null_message_handler,
+                                          10,
+                                          true,
+                                          {},
+                                          string_preprocess,
+                                          class_hierarchy,
+                                          false};
+
+  GIVEN("An int_array variable")
+  {
+    const source_locationt location;
+    const typet int_array_type = java_array_type('i');
+    const symbol_exprt int_array{"int_array", int_array_type};
+    const std::size_t variable_index = 0;
+    java_bytecode_convert_method_unit_testt::add_variable(
+      converter, variable_index, int_array, 0, 1, false, {});
+    const std::size_t address = 0;
+    WHEN("variable is called as it would be for an `aload` instruction")
+    {
+      const exprt index_expr = from_integer(variable_index, java_int_type());
+      const exprt result = java_bytecode_convert_method_unit_testt::variable(
+        converter, index_expr, 'a', address, true);
+      THEN("the result is int_array")
+      {
+        REQUIRE(result == int_array);
+      }
+    }
+    WHEN("There is no variable at the given index")
```

Comment:
:bulb: Ideally we'd test a non-array reference type, and a primitive type (not sure what the latter should do, maybe throw?)